docs/writing-rules/experiments/pattern-syntax.md

```diff
@@ -208,6 +208,37 @@ where:
   - focus: $A
 ```
 
+## <i class="fa-solid fa-exclamation"></i> `as-metavariable`
+
+`as-metavariable` is a rule-writing feature that bridges the gap between metavariables and matches. Metavariables get access to things like `metavariable-comparison`, `metavariable-regex`, and `metavariable-pattern`, but you can’t use them on arbitrary matches. However, the `as` operator lets you embed arbitrary matches into metavariables, or bind arbitrary matches to a name.
+
+The syntax is as follows:
+
+```yaml
+all:
+ - pattern: |
+ @decorator
+ def $FUNC(...):
+ ...
+    as: $DECORATED_FUNC
```

Comment:
this `as` is one character too far up, it should be vertically aligned with the `p` in `pattern`